src/format.py

```diff
@@ -121,42 +128,47 @@ def getSelected(self):
         return self.selected
 
     def getBefore(self):
-        return self.line[0:self.start]
+        return str(self.formattedLine)[0:self.start]
 
     def getAfter(self):
-        return self.line[self.end:]
+        return str(self.formattedLine)[self.end:]
 
     def getMatch(self):
         return self.group
 
     def __str__(self):
-        return self.getBefore() + '||' + self.getMatch(
-        ) + '||' + self.getAfter() + '||' + str(self.num)
-
-    def getStyleForState(self):
-        curses.init_pair(1, curses.COLOR_WHITE, curses.COLOR_RED)
-        curses.init_pair(2, curses.COLOR_WHITE, curses.COLOR_BLUE)
-        curses.init_pair(3, curses.COLOR_WHITE, curses.COLOR_GREEN)
+        return (self.getBefore() + '||' + self.getMatch()
+                + '||' + self.getAfter() + '||' +
+                str(self.num))
 
+    def getDecoratorFormatting(self):
         if self.hovered and self.selected:
-            return curses.color_pair(3)
+            attributes = (curses.COLOR_WHITE, curses.COLOR_RED, 0)
```

Comment:
oooh nice, this is way more straight-forward than the hacky init_pair stuff I was doing :tada: